activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -401,14 +401,8 @@ def attributes_for_create(attribute_names)
       def format_for_inspect(value)
         if value.is_a?(String) && value.length > 50
           "#{value[0, 50]}...".inspect
-        elsif value.is_a?(Time)
-          if Time::DATE_FORMATS[:db].respond_to?(:call)
-            Time::DATE_FORMATS[:db].call(value)
-          else
-            %("#{value.strftime(Time::DATE_FORMATS[:db] + '.%9N')}")
-          end
-        elsif value.is_a?(Date)
-          %("#{value.to_s(:db)}")
+        elsif value.is_a?(Date) || value.is_a?(Time)
+          %("#{value.to_s(:inspect)}")
```

Comment:
`to_s(:inspect)` looks odd. I think simpler solution to see nsec is just use `value.inspect`.

Reply:
Yeah. I agree with @kamipo. Adding new option will cause a drift between the `:db` and `:inspect` formats if people are changing the `:db`. At that point, why bother on trying to inspect with a different format that is not `value.inspect`?

Reply:
Since we already drifted away from the `:db` format by adding nsec, I assumed the reason we continued to use `value.to_s(:xxxx)` instead of `value.inspect` was configurability.  I'm happy to submit another PR that changes this to `value.inspect` though.

Reply:
Attribute values are typically instances of `ActiveSupport::TimeWithZone`, which responds true to `is_a?(Time)` but has a different inspect format:

```ruby
irb(main):001:0> Time.now.inspect
=> "2020-05-07 02:11:46.798268 +0100"
irb(main):002:0> Time.now.in_time_zone(Time.find_zone!("UTC")).inspect
=> "Thu, 07 May 2020 01:11:47.963176000 UTC +00:00"
irb(main):003:0> Time.now.in_time_zone(Time.find_zone!("UTC")).to_s(:inspect)
=> "2020-05-07 01:11:48.786426000 +0000"
```

That format is too long to use here, in my opinion. We could add a special case for TimeWithZone, but at that point the implementation in this PR seems like a simpler option.